addons/mail/models/mail_message.py

```diff
@@ -588,9 +588,12 @@ def _validate_access_for_current_persona(self, operation):
         if not self:
             return False
         self.ensure_one()
-        self.sudo(False).check_access_rule(operation)
-        self.sudo(False).check_access_rights(operation)
-        return True
+        try:
+            self.sudo(False).check_access_rule(operation)
+            self.sudo(False).check_access_rights(operation)
+            return True
+        except AccessError:
+            return False
```

Comment:
There is a single call to this method, the code can probably be moved there.
But I think it might be ok to fully drop it in favor of using just `_get_thread_with_access`.